src/pudl_archiver/archivers/ferc/xbrl.py

```diff
@@ -108,11 +115,47 @@ def parse_timestamp(cls, timestamp: time.struct_time):  # noqa: N805
         return datetime.datetime.fromtimestamp(time.mktime(tuple(timestamp)))
 
     def __hash__(self):
-        """Implement hash so FeedEntry can be used in a set.
+        """Implement hash so FeedEntry can be used in a set."""
+        return hash(f"{self.download_url}")
 
-        Entry ID's are unique, so that's all that is needed for the hash.
-        """
-        return hash(f"{self.entry_id}")
+    def __eq__(self, other: "FeedEntry"):
+        """Implement eq so FeedEntry can be used in a set."""
+        return self.download_url == other.download_url
+
+
+def _taxonomy_zip_name_from_url(url: str) -> str:
+    if not (match := TAXONOMY_URL_PATTERN.match(url)):
+        raise RuntimeError(f"{url} does not appear to be a taxonomy url.")
+    return f"{match.group(1).replace('_', '-')}.zip"
+
+
+class FilingMetadata(BaseModel):
+    """Combines RSS feed metadata with taxonomy referenced in filing."""
+
+    filename: str
+    rss_metadata: FeedEntry
+    taxonomy_url: str
+    taxonomy_zip_name: str
+
+    @classmethod
+    def from_rss_metadata(
+        cls, rss_metadata: FeedEntry, filename: str, filing_data: bytes
+    ) -> "FilingMetadata":
+        """Construct metadata from RSS feed and filing data to extract taxonomy URL."""
+        if not (match := TAXONOMY_URL_PATTERN.search(filing_data.decode().lower())):
+            raise RuntimeError(
+                f"Couldn't find taxonomy for filing {rss_metadata.download_url}"
+            )
+
+        taxonomy_url = match.group(0)
+        return cls(
+            filename=f"{rss_metadata.title}_form{rss_metadata.ferc_formname.as_int()}_{rss_metadata.ferc_period}_{round(rss_metadata.published_parsed.timestamp())}.xbrl".replace(
```

Comment:
nit: this specific string formatting thing would be clearer as a separate function - or the `filename` can just be an `@property`.

Reply:
Oh yeah, I actually wanted to just pass the already formatted `filename` in and then use that here instead of formatting again. This is used by the extractor, so I do need the `filename` in metadata